master/buildbot/process/buildstep.py

```diff
@@ -566,6 +560,14 @@ def setRenderable(res, attr):
             # we describe ourselves only when renderables are interpolated
             self.realUpdateSummary()
 
+            # check doStepIf (after rendering)
```

Comment:
You are  right, converting str/unicode is not necessary. 
I removed the original two commits, a new one is below.

About unit tests, I agree, but I have no idea about howto write them. So I will not implement. Sorry.